### PR TITLE
515 - ac - Comparison DTI

### PR DESCRIPTION
**Pull request by annacanal**
- [x] ComparisonDTI
- [x] test_ComparisonDTI

### CODE REVIEW

File: braph2/workflows/DTI/ComparisonDTI.m [hunk 1]
```diff
@@ -0,0 +1,341 @@
...
+                    repmat({0}, 1, group_2.subjectnumber()), ...
+                    'ComparisonDTI.values_2', ...
+                    varargin{:});
+                assert(iscell(c.getGroupValue(1)) & ...
+                    isequal(size(c.getGroupValue(1)), [1, group_1.subjectnumber]) & ...
+                    all(cellfun(@(x) isequal(size(x), [1, 1]), c.getGroupValue(1))), ...
+                    [BRAPH2.STR ':ComparisonDTI:' BRAPH2.WRONG_INPUT], ...
+                    ['Data not compatible with ComparisonDTI.'])  %#ok<*NBRAK>
+                assert(iscell(c.getGroupValue(2)) & ...
+                    isequal(size(c.getGroupValue(2)), [1, group_1.subjectnumber]) & ...
```
Comment (egolol) [inline braph2/workflows/DTI/ComparisonDTI.m:90]: group2

File: braph2/workflows/DTI/ComparisonDTI.m [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+                    [BRAPH2.STR ':ComparisonDTI:' BRAPH2.WRONG_INPUT], ...
+                    ['Data not compatible with ComparisonDTI.'])
+                
+            elseif Measure.is_binodal(measure_code)  % binodal measure
+                c.values_1 = get_from_varargin( ...
+                    repmat({zeros(atlas.getBrainRegions().length())}, 1, group_1.subjectnumber()), ...
+                    'ComparisonDTI.values_1', ...
+                    varargin{:});
+                c.values_2 = get_from_varargin( ...
+                    repmat({zeros(atlas.getBrainRegions().length())}, 1, group_1.subjectnumber()), ...
```
Comment (egolol) [inline braph2/workflows/DTI/ComparisonDTI.m:236]: grp2

File: braph2/workflows/DTI/ComparisonDTI.m [hunk 3]
```diff
@@ -0,0 +1,341 @@
...
+                    [BRAPH2.STR ':ComparisonDTI:' BRAPH2.WRONG_INPUT], ...
+                    ['Data not compatible with ComparisonDTI.'])
+                
+            elseif Measure.is_nodal(measure_code)  % nodal measure
+                c.values_1 = get_from_varargin( ...
+                    repmat({zeros(atlas.getBrainRegions().length(), 1)}, 1, group_1.subjectnumber()), ...
+                    'ComparisonDTI.values_1', ...
+                    varargin{:});
+                c.values_2 = get_from_varargin( ...
+                    repmat({zeros(atlas.getBrainRegions().length(), 1)}, 1, group_1.subjectnumber()), ...
```
Comment (egolol) [inline braph2/workflows/DTI/ComparisonDTI.m:156]: grp2

File: braph2/workflows/DTI/ComparisonDTI.m [hunk 4]
```diff
@@ -0,0 +1,341 @@
...
+        end
+        function subject_class = getSubjectClass()
+            subject_class = 'SubjectDTI';
+        end
+        function available_settings = getAvailableSettings()
+            available_settings = {
+                'ComparisonDTI.PermutationNumber', BRAPH2.NUMERIC, 1000, {};
+                };
+        end
+        function sub = getComparison(comparisonClass, id, label, notes, atlas, group_1, group_2, varargin) %#ok<INUSD>
```
Comment (egolol) [inline braph2/workflows/DTI/ComparisonDTI.m:337]: measure_code missing

Comment (annacanal) [reply braph2/workflows/DTI/ComparisonDTI.m:337]: True.
This structure was according to the ComparisonMRI, so it will have to be updated there also

**Pull request merged**